applications/decentralized-security-marketplace.md

```diff
@@ -0,0 +1,278 @@
+
+![Successfull report](https://github.com/magic-powered/decentralised-security-marketplace-architecture/blob/main/img/successfull-report.png?raw=true)
+
+The judge can change the reward size in the proposal and the serenity of found issues at any time by consensus mechanism together with other judges.
+
+![Change report submission](https://github.com/magic-powered/decentralised-security-marketplace-architecture/blob/main/img/change-proposal.png?raw=true)
+
+Any eligible community member can become a judge. Judges are rewarded from the same rewards pool with a percent of rewards to be distributed by the proposals they review.
+
+The eligibility criteria for a community member to become a judge will be provided during the protocol design phase.
+
+### Project Details
+
+The central piece of the platform will be the smart contracts:
+
+- **Rewards pool contract**: Lock and release rewards funds for developers that make audits.
+- **Governance contract**: Control and verify audit reports submissions
+- **Souldbound NFT**: Help identify if a community member is a Judge and can vote for audits reports. - the protocol is not aiming to implement this, the protocol will support standard nft
+
+![Smart contract composition](https://github.com/magic-powered/decentralised-security-marketplace-architecture/blob/main/img/architecture/smart-contracts.png?raw=true)
+
+#### Preliminary interface of the smart contracts
+
+* Note that this list preliminary interface is hacky and dirty. It is provided to show what our team has in mind
+
+##### Rewards Pool
+
+```rust
+mod rewardspool {
+    pub type ProjectId = u32;
+    pub type ProposalId = u32;
+
+    pub struct RewardsPool {
+        /// TODO
+    }
+    
+    pub enum Error {
+        /// TODO
+    }
+    
+    pub enum BountyType {
+        /// TODO
+    }
+    
+    pub enum ProjectStatus {
+        Pending,
+        Started,
+        Completed
+    }
+    
+    pub enum ProjectPolicy {
+        AdminOnly,
+        JudgesOnly,
+        AdminAndJudges
+    }
+    
+    impl RewardsPool {
+        /// create an empty project. Recently created project will have a status: Pending
+        pub fn create_project(&mut self, id: ProjectId) -> ProjectId {}
+        /// defines whatever who can vote for rewards proposals
+        pub fn set_project_policy(&mut self, id: ProjectId, policy: ProjectPolicy, numberOfVotes: u8) -> Result<(), Error> {}
+        /// get project policy. Called mainly by governance contract 
+        pub fn get_project_policy(&mut self, id: ProjectId) -> (ProjectPolicy, u8) {}
+        /// append bounty for specific job type. job type later can be used to create proposal for reward
+        pub fn append_bounty(&mut self, id: ProjectId, bountyType: BountyType, reward: u32) -> Result<(), Error> {}
```

Comment:
So single project can have multiple bounties. 
In the case of security audits it could be different serenity levels of the issues found:

P0 - critical issue - possible money or ownership loss - 1000 USD
P1 - major issue - any bugs preventing expected business logic execution - 500 USD
P3 - minor issues - any other minor issues found - 100 USD

different project may have different set of bounties so bounty type is a formal identification of the type of work that needs to be done while project is on going state and append means that users can create their sets of bounties within their projects